src/main/java/uk/gov/hmcts/darts/automation/utils/Postgres.java

```diff
@@ -39,110 +46,259 @@ public Postgres() {
+	}
+
+	public ResultSet executeSql(String sql) throws Exception {
+		connect();
+		try (Statement stmt = conn.createStatement();
+				ResultSet rs = stmt.executeQuery(sql)) {
+			return rs;
+		}
+	}
+
+	public int deleteRow(String table, String key, String keyVal) throws Exception {
+		log.info("Database delete: " + table + " " + key + " " + keyVal);
+		String sql = "delete from " + table + " where " + key + " = " +  delimitedValue(table, key, keyVal);
+		return deleteRow(sql);
+	}
+
+	public int deleteRow(String sql) throws Exception {
+		int deleteCount = 0;
+		connect();
+		conn.setAutoCommit(false);
+		log.info("Database delete: " + sql);
+		try (PreparedStatement stmt = conn.prepareStatement(sql)) {
+			deleteCount = stmt.executeUpdate();
+			if (deleteCount == 1) {
+				conn.commit();
+			} else {
+				conn.rollback();
+				if (deleteCount == 0) {
+					throw new Exception("No rows deleted");
+				} else {
+					throw new Exception("Excess rows deleted " + deleteCount);
+				}
+			}
+		}
+		return deleteCount;
+	}
 	
-			
+	String delimitedValue(String table, String column, String value) throws Exception {
+		String columnType = returnSingleValue("select pg_typeof(" + column + ") "
+				+ "from " + table
+				+ " LIMIT 1");
+		String delimiter = "";
+		if (value == null || value.equalsIgnoreCase("null")) {
+			value = "null";
+		} else {
+			switch(columnType.split(" ")[0].toLowerCase()) {
+			case "character": 
+			case "string": 
+			case "date": 
+					delimiter = "'";
+					break;
+			default : delimiter = "";
+			}
+		}
+		return delimiter + value + delimiter;
+	}
+
+	public int deleteRows(String sql) throws Exception {
+		int deleteCount = 0;
+		connect();
+		conn.setAutoCommit(false);
+		log.info("Database delete: " + sql);
+		try (PreparedStatement stmt = conn.prepareStatement(sql)) {
+			deleteCount = stmt.executeUpdate();
+			if (deleteCount == 0) {
+				conn.rollback();
+				throw new Exception("No rows deleted");
+			} else {
+				conn.commit();
+				log.info("Rows deleted " + deleteCount);
+			}
+		}
+		return deleteCount;
+	}
+
+	public int updateRow(String table, String keyCol, String keyVal, String UpdateCol, String newVal) throws Exception {
+		log.info("Database update: " + table + " " + keyCol + " " + keyVal + " " + UpdateCol + " " + newVal);
+		String sql = "update darts." + table 
+				+ " set " + UpdateCol + " = " + delimitedValue(table, UpdateCol, newVal)
+				+ " where " + keyCol + " = "  + delimitedValue(table, keyCol, keyVal);
+		int updateCount = 0;
+		connect();
+		conn.setAutoCommit(false);
+		log.info("Database update: " + sql);
+		try (PreparedStatement stmt = conn.prepareStatement(sql)) {
+			updateCount = stmt.executeUpdate();
+			if (updateCount == 1) {
+				conn.commit();
+			} else {
+				conn.rollback();
+				if (updateCount == 0) {
+					throw new Exception("No rows updated");
+				} else {
+					throw new Exception("Excess rows updated " + updateCount);
+				}
+			}
+		}
+		return updateCount;
+	}
```

Comment:
Would it make sense, here and elsewhere it happens in the code, once the SQL string for "updateRow" has been constructed to call the other "updateRow" method that expects a fully constructed SQL string? 
[To help avoid potential code duplication.]

Reply:
yep
Done